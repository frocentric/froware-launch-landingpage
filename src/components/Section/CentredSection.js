import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Title = styled.h1`
  text-align: center;
`;

const Container = styled.div`
  grid-column: 2/12;
  width: 506px;
`;

const CentredSection = ({ title, body }) => (
  <Container>
    <Title>{title}</Title>
    {body.map((text, index) => (
      <p key={`${index}-${title}`} data-testid='section-text'>
        {text}
      </p>
    ))}
  </Container>
);

CentredSection.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.array.isRequired,
};

export default CentredSection;
