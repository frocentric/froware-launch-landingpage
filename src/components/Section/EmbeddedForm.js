import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  grid-column: 2/12;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 506px;
`;

const Text = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 316px;
`;

const Children = styled.div`
  width: 371px;
  height: 226px;
  background: #c4c4c4;
  border-radius: 10px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const EmbeddedForm = ({ title, body, children }) => (
  <Container>
    <Text>
      <h1>{title}</h1>
      {body.map((text, index) => (
        <p key={`${index}-${title}`} data-testid='section-text'>
          {text}
        </p>
      ))}
    </Text>
    <Children>{children}</Children>
  </Container>
);

EmbeddedForm.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.array.isRequired,
  children: PropTypes.node,
};

export default EmbeddedForm;
