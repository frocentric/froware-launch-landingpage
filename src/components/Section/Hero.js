import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  display: grid;
  grid-template-columns: repeat(12, 1fr);

  @media (min-width: 1024px) {
    height: 60%;
  }
  @media (min-width: 1440px) {
    height: 80%;
  }
`;

const Image = styled.img`
  grid-column: 2/12;
  width: 100%;
  height: auto;
  margin-top: 40px;

  @media (min-width: 1024px) {
    grid-area: 1 / 1 / 2 / 9;
    margin-top: 0;
    align-self: end;
  }
`;

const TextWrapper = styled.div`
  grid-column: 2/12;
  text-align: center;

  @media (min-width: 1024px) {
    grid-area: 1 / 8 / 2 / 12;
    text-align: justify;
    align-self: start;
  }

  @media (min-width: 1440px) {
    grid-area: 1 / 8 / 2 / 12;
    text-align: justify;
    margin-top: 40px;
  }
`;

const Title = styled.h1`
  text-align: centre;

  @media (min-width: 1024px) {
    text-align: left;
  }
`;

const CentredSection = ({ title, body, image }) => (
  <Container>
    <Image src={image} alt={title} />
    <TextWrapper>
      <Title>{title}</Title>
      {body.map((text, index) => (
        <p key={`${index}-${title}`} data-testid='section-text'>
          {text}
        </p>
      ))}
    </TextWrapper>
  </Container>
);

CentredSection.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.array.isRequired,
};

export default CentredSection;
