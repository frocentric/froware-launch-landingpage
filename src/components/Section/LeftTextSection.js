import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const TextWrapper = styled.div`
  grid-column: 2/7;
`;

const Image = styled.img`
  grid-column: 7/12;
`;

const LeftTextSection = ({ title, body, image }) => (
  <>
    <TextWrapper>
      <h1>{title}</h1>
      {body.map((text, index) => (
        <p key={`${index}-${title}`}>{text}</p>
      ))}
    </TextWrapper>
    <Image src={image} alt={title} />
  </>
);

LeftTextSection.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string.isRequired,
  body: PropTypes.array.isRequired,
};

export default LeftTextSection;
