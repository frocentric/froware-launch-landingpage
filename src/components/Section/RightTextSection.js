import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const TextWrapper = styled.div`
  grid-column: 2/12;

  @media (min-width: 1024px) {
    grid-column: 6/12;
  }
`;

const Image = styled.img`
  grid-column: 2/12;
  max-width: 100%;
  padding-top: 30px;

  @media (min-width: 1024px) {
    grid-column: 2/6;
    padding: 80px 0;
  }
`;

const RightTextSection = ({ title, body, image }) => {
  return (
    <>
      <Image src={image} alt={title} />
      <TextWrapper>
        <h1>{title}</h1>
        {body.map((text, index) => (
          <p key={`${index}-${title}`} data-testid='section-text'>
            {text}
          </p>
        ))}
      </TextWrapper>
    </>
  );
};

RightTextSection.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string.isRequired,
  body: PropTypes.array.isRequired,
};

export default RightTextSection;
