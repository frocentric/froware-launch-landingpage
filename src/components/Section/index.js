import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Hero from './Hero';
import RightTextSection from './RightTextSection';
import LeftTextSection from './LeftTextSection';
import EmbeddedForm from './EmbeddedForm';
import Button from '../Button';

const handleBackgroundColour = (alignment) => {
  switch (alignment) {
    case 'left':
      return '#EBEBEB';
    case 'hero':
      return '#f4f4f4';
    default:
      return '#fff';
  }
};

const Container = styled.div`
  display: grid;
  grid-template-columns: repeat(12, 1fr);

  @media (min-width: 1024px) {
    height: 100vh;
  }
`;

const Wrapper = styled.div`
  background-color: ${({ alignment }) => handleBackgroundColour(alignment)};
  grid-column: 1/-1;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  flex-direction: column;
`;

const Section = React.forwardRef(({ section, onClick, id }, ref) => {
  const { alignment, image, title, body } = section;

  const SECTIONS = {
    hero: <Hero title={title} body={body} image={image} />,
    left: <LeftTextSection title={title} body={body} image={image} />,
    right: <RightTextSection title={title} body={body} image={image} />,
    none: <EmbeddedForm title={title} body={body} />,
  };
  
  return (
    <Container ref={ref}>
      <Wrapper alignment={alignment}>
        {SECTIONS[alignment]}{' '}
        {alignment !== 'none' && <Button onClick={onClick} sectionId={id} />}
      </Wrapper>
    </Container>
  );
});

Section.propTypes = {
  sections: PropTypes.shape({
    alignment: PropTypes.string.isRequired,
    image: PropTypes.string,
    title: PropTypes.string.isRequired,
    body: PropTypes.array.isRequired,
  }),
};

export default Section;
