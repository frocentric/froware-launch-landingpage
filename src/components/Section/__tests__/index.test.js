import React from 'react';
import { render } from '@testing-library/react';
import Section from '../';
import { sections } from '../../../fixtures/sections';

describe('<Section />', () => {
  it('should render a section', () => {
    const section = render(<Section section={sections[1]} />);

    expect(section).toMatchSnapshot();
  });
});
