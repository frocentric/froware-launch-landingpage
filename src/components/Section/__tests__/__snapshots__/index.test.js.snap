// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should render a section 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jrsJCI evakZi"
      >
        <div
          class="sc-kEqYlL heKzEZ"
        >
          <img
            alt="Test Title 2"
            class="sc-jSFkmK kLbfTo"
            src="awareness.svg"
          />
          <div
            class="sc-eCApGN bwVyXU"
          >
            <h1>
              Test Title 2
            </h1>
            <p
              data-testid="section-text"
            >
              test body 1
            </p>
            <p
              data-testid="section-text"
            >
              test body 2
            </p>
          </div>
           
          <div
            class="sc-pNWxx crXZMz"
          >
            <svg
              fill="none"
              height="54"
              viewBox="0 0 54 54"
              width="54"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0)"
              >
                <path
                  d="M52.7401 18.3601C56.8801 31.1401 51.1201 45.1801 39.0601 51.1201C37.9801 51.6601 36.7201 52.2001 35.6401 52.5601C22.8601 56.8801 8.82009 51.1201 2.88009 39.0601C2.16008 37.9801 1.80009 36.7201 1.44008 35.4601C-2.69992 22.6801 3.06008 8.64007 15.1201 2.70007C16.2001 2.16007 17.4601 1.62007 18.5401 1.26007C31.3201 -2.87993 45.3601 2.88007 51.3001 14.9401C51.8401 16.0201 52.3801 17.1001 52.7401 18.3601ZM4.50008 23.9401C2.88008 36.5401 11.5201 47.8801 24.1201 49.5001C36.7201 51.1201 48.0601 42.3001 49.6801 29.8801C51.3001 17.4601 42.4801 5.94007 30.0601 4.32007C28.0801 4.14007 26.1001 4.14007 24.1201 4.32007C13.8601 5.76007 5.76008 13.6801 4.50008 23.9401ZM37.0801 21.9601L27.0001 32.0401L16.9201 21.9601C16.0201 21.0601 14.7601 21.0601 14.0401 21.9601C13.1401 22.8601 13.1401 24.1201 14.0401 25.0201L27.0001 37.9801L40.1401 24.8401C41.0401 23.9401 41.0401 22.6801 40.1401 21.7801C39.2401 21.0601 37.9801 21.0601 37.0801 21.9601Z"
                  fill="black"
                />
              </g>
              <defs>
                <clippath
                  id="clip0"
                >
                  <rect
                    fill="white"
                    height="54"
                    transform="translate(0 54) rotate(-90)"
                    width="54"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jrsJCI evakZi"
    >
      <div
        class="sc-kEqYlL heKzEZ"
      >
        <img
          alt="Test Title 2"
          class="sc-jSFkmK kLbfTo"
          src="awareness.svg"
        />
        <div
          class="sc-eCApGN bwVyXU"
        >
          <h1>
            Test Title 2
          </h1>
          <p
            data-testid="section-text"
          >
            test body 1
          </p>
          <p
            data-testid="section-text"
          >
            test body 2
          </p>
        </div>
         
        <div
          class="sc-pNWxx crXZMz"
        >
          <svg
            fill="none"
            height="54"
            viewBox="0 0 54 54"
            width="54"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0)"
            >
              <path
                d="M52.7401 18.3601C56.8801 31.1401 51.1201 45.1801 39.0601 51.1201C37.9801 51.6601 36.7201 52.2001 35.6401 52.5601C22.8601 56.8801 8.82009 51.1201 2.88009 39.0601C2.16008 37.9801 1.80009 36.7201 1.44008 35.4601C-2.69992 22.6801 3.06008 8.64007 15.1201 2.70007C16.2001 2.16007 17.4601 1.62007 18.5401 1.26007C31.3201 -2.87993 45.3601 2.88007 51.3001 14.9401C51.8401 16.0201 52.3801 17.1001 52.7401 18.3601ZM4.50008 23.9401C2.88008 36.5401 11.5201 47.8801 24.1201 49.5001C36.7201 51.1201 48.0601 42.3001 49.6801 29.8801C51.3001 17.4601 42.4801 5.94007 30.0601 4.32007C28.0801 4.14007 26.1001 4.14007 24.1201 4.32007C13.8601 5.76007 5.76008 13.6801 4.50008 23.9401ZM37.0801 21.9601L27.0001 32.0401L16.9201 21.9601C16.0201 21.0601 14.7601 21.0601 14.0401 21.9601C13.1401 22.8601 13.1401 24.1201 14.0401 25.0201L27.0001 37.9801L40.1401 24.8401C41.0401 23.9401 41.0401 22.6801 40.1401 21.7801C39.2401 21.0601 37.9801 21.0601 37.0801 21.9601Z"
                fill="black"
              />
            </g>
            <defs>
              <clippath
                id="clip0"
              >
                <rect
                  fill="white"
                  height="54"
                  transform="translate(0 54) rotate(-90)"
                  width="54"
                />
              </clippath>
            </defs>
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
